lib/watir/elements/select.rb

```diff
@@ -161,6 +196,23 @@ def select_by(how, str_or_rx)
       select_matching(found)
     end
 
+    def select_by!(how, str_or_rx, number)
+      script = how == :text ? text_script : value_script
+
+      if str_or_rx.is_a? Regexp
+        js_rx = str_or_rx.inspect.sub('\\A','^').sub('\\Z','$').sub('\\z','$').sub(/^\//,'').sub(/\/[a-z]*$/,'').gsub(/\(\?#.+\)/, '').gsub(/\(\?-\w+:/,'(')
```

Comment:
Oh yes, that's what I was reading. That being said, I feel like that's why we're doing these with bangs. They are a slimmed down, use at your own risk kind of thing. Hopefully others who find this interesting can improve on it. This is just the extent of my JS and/or Stack Overflow searching capabilities. :)